cryptography/hedera-cryptography-altbn128/src/main/rust/group_element_utils.rs

```diff
@@ -99,3 +105,44 @@ pub fn group_elements_deserialize_and_validate<G: CurveGroup>(
         Err(err) => Err(err.to_string()),
     }
 }
+
+/// returns the x coordinate of a point if the point is in the curve
+pub fn x_coordinate_from_hash<G: SWCurveConfig>( candidate_hash: &[u8])->Option<G::BaseField>{
+    if G::BaseField::extension_degree() == 1 {
+        // if the order is 1 the op is simpler
+        let f = <G::BaseField as Field>::BasePrimeField::from_be_bytes_mod_order(
+            &candidate_hash,
+        );
+        G::BaseField::from_base_prime_field_elems(&[f])
+    } else {
+        G::BaseField::from_random_bytes(&candidate_hash)
+    }
+}
```

Comment:
This method is more generic than its name suggests. It takes a candidate byte array and returns a BaseField representation. The base field element can then be used as an x or y coordinate, but that's up to the caller.

```suggestion
/// Converts a u8 array into the BaseField representation of the group (the field where the coordinates are represented) if there is such representation
pub fn coordinate_from_hash<G: SWCurveConfig>( candidate: &[u8])->Option<G::BaseField>{
    if G::BaseField::extension_degree() == 1 {
        // if the order is 1 the op is simpler
        let f = <G::BaseField as Field>::BasePrimeField::from_be_bytes_mod_order(
            &candidate,
        );
        G::BaseField::from_base_prime_field_elems(&[f])
    } else {
        G::BaseField::from_random_bytes(&candidate)
    }
}
```